src/h5web/nexus-visualizer/NexusVisualizer.tsx

```diff
@@ -1,17 +1,70 @@
 import React, { ReactElement } from 'react';
 import type { HDF5Group } from '../providers/models';
+import { getSignalLink, getInterpretationVis } from './utils';
 import styles from '../Visualizer.module.css';
+import { useEntity } from '../providers/hooks';
+import { isDataset } from '../providers/utils';
+import VisSelector from '../dataset-visualizer/VisSelector';
+import VisDisplay from '../dataset-visualizer/VisDisplay';
+import { VIS_DEFS } from '../visualizations';
+import { getSupportedVis } from '../dataset-visualizer/utils';
 
 interface Props {
   group: HDF5Group;
 }
 
 function NexusVisualizer(props: Props): ReactElement {
   const { group } = props;
+
+  const signalLink = getSignalLink(group);
+  const signalDataset = useEntity(signalLink);
+
+  if (!signalLink) {
+    return (
+      <div className={styles.visualizer}>
+        <div className={styles.visBar} />
+        <p className={styles.fallback}>
+          NXdata group has no attribute signal or it points to an non-existing
+          dataset.
+        </p>
+      </div>
+    );
+  }
+
+  if (!signalDataset || !isDataset(signalDataset)) {
+    return (
+      <div className={styles.visualizer}>
+        <div className={styles.visBar} />
+        <p className={styles.fallback}>
+          NXdata signal {signalLink.title} is not a valid dataset.
+        </p>
+      </div>
+    );
+  }
+
+  // Prefer vis deduced from interpretation. Else take the first supported vis.
+  const nexusVis =
+    getInterpretationVis(group) ?? getSupportedVis(signalDataset)[0];
```

Comment:
Jumping the gun a bit here 😄 - oh well, it works. We'll have to remember to add some sort of error recovery code, if the interpretation doesn't match the dimensions of the signal dataset for instance.

Reply:
`getSupportedVis(signalDataset)[0]` will always return `Vis.Matrix` for nD datasets. You want to get the last one in the array, I think.